### DISCUSSION TITLE
How do i get an ItemStack/Material from a Block?

### DISCUSSION THREAD

**Discussion by MrGazdag** [Minestom/Minestom#735] [Help] [answered]
If I have a Block instance, how would I get an ItemStack or a Material instance from that? Is there a built-in way to get this information or do I have to map every block to its respective item manually?

**Comment by MrlnHi** [comment, marked as answer]
Answer stolen from TheMode, respectfully (read comments of this answer to understand why):
```java
Block.STONE.registry().material();
```

**Reply by MrlnHi** [reply]
@MrGazdag please mark TheMode's answer as correct xd

**Reply by MrGazdag** [reply]
i seem to not be able to do that unfortunately @Marlonzockt, could you edit your answer to include TheMode's answer (with credit of course)?

**Comment by TheMode** [comment]
```java
Block.STONE.registry().material()
```
